hypothesis-extra/hypothesis-pytest/src/hypothesispytest/__init__.py

```diff
@@ -19,17 +19,20 @@
 
 class StoringReporter(object):
 
-    def __init__(self):
+    def __init__(self, config):
+        self.config = config
         self.results = []
 
     def __call__(self, msg):
+        if self.config.getoption('capture', 'fd') == 'no':
+            print(msg)
```

Comment:
This should delegate to another reporter rather than printing, which might not be the correct thing to do depending on what is installed (although I don't think in practice this is ever a problem right now). But e.g. if the test were being run inside a with with_reporter(reporting.silent) this would do the wrong thing. 

The relevant machinery is in https://github.com/DRMacIver/hypothesis/blob/master/src/hypothesis/reporting.py